packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flutter/widgets.dart';
+
+enum Direction { horizontal, vertical }
+
+/// Super class for [RowComponent] and [ColumnComponent]
+/// A relayout is performed when
+///  - a change in this component's children takes place
+///  - the [gap] parameter is changed
+abstract class LayoutComponent extends PositionComponent {
+  LayoutComponent(
+    this.direction,
+    this.alignment,
```

Comment:
So this should be removed?

Reply:
No no, the gap still works since it leads to an additional growing of the size.
But the spaceXY modes calculate the available space of the existing components and since there is no free space to be distributed they don't have an effect.

Reply:
Not the gap, the alignment (comments refer to the line above)

Reply:
Some alignment modes still work, e.g. start and end and possibly center as well.

Reply:
Then we should use an enum of our own for this that only includes the modes that we support.